tests/ballerina-test/src/test/java/org/ballerinalang/test/collections/VectorNegativeTest.java

```diff
@@ -0,0 +1,179 @@
+
+        Assert.assertNotNull(returns);
+
+        BStruct vector = (BStruct) returns[0];
+        BRefValueArray vectorEntries = (BRefValueArray) vector.getRefField(0);
+        long finalVectorSize = vector.getIntField(0);
+
+        Assert.assertEquals(finalVectorSize, expectedFinalValues.length);
+
+        for (int i = 0; i < finalVectorSize; i++) {
+            Assert.assertEquals(vectorEntries.get(i).value(), expectedFinalValues[i]);
+        }
+
+        BRefValueArray removedVals = (BRefValueArray) returns[1];
+        Assert.assertEquals(removedVals.size(), 1); // only 1 valid index
+        Assert.assertEquals(((BInteger) removedVals.get(0)).intValue(), 10);
+
+        BRefValueArray errors = (BRefValueArray) returns[2];
+        Assert.assertEquals(errors.size(), expectedErrMsgs.length);
+
+        for (int i = 0; i < expectedErrMsgs.length; i++) {
+            Assert.assertEquals(((BStruct) errors.get(i)).getStringField(0), expectedErrMsgs[i]);
+        }
+    }
+
+    @Test(description = "Test case for testing replacement of elements outside the bounds of a vector.")
+    public void testReplaceIndexOutOfRange() {
+        long[] values = new long[]{100, 110, 120};
+        long[] indices = new long[]{3, 10, -1}; // 1 valid index and 2 invalid indices
+        int vectorSize = 10;
+        long[] expectedFinalValues = new long[]{10, 20, 30, 100, 50, 60, 70, 80, 90, 100};
+        String[] expectedErrMsgs = new String[]{"Index out of range: 10", "Index out of range: -1"};
+
+        BValue[] returns = BRunUtil.invoke(compileResult, "testReplaceIndexOutOfRange",
+                                           new BValue[]{buildIntArray(values), buildIntArray(indices),
+                                                   new BInteger(vectorSize)});
+
+        Assert.assertNotNull(returns);
+
+        BStruct vector = (BStruct) returns[0];
+        BRefValueArray vectorEntries = (BRefValueArray) vector.getRefField(0);
+        long finalVectorSize = vector.getIntField(0);
+
+        Assert.assertEquals(finalVectorSize, vectorSize);
+
+        for (int i = 0; i < expectedFinalValues.length; i++) {
+            Assert.assertEquals(vectorEntries.get(i).value(), expectedFinalValues[i]);
+        }
+
+        BRefValueArray errors = (BRefValueArray) returns[1];
+
+        Assert.assertEquals(errors.size(), 3); // Since there are 3 insertions
+        Assert.assertNull(errors.get(0)); // Since first insertion is valid, the error is null
+        Assert.assertEquals(((BStruct) errors.get(1)).getStringField(0), expectedErrMsgs[0]);
+        Assert.assertEquals(((BStruct) errors.get(2)).getStringField(0), expectedErrMsgs[1]);
+    }
+
+    private BIntArray buildIntArray(long[] args) {
+        BIntArray valueArray = new BIntArray();
+
+        for (int i = 0; i < args.length; i++) {
+            valueArray.add(i, args[i]);
+        }
+
+        return valueArray;
+    }
+}
```

Comment:
@pubudu91 Should we have a test for size function ?

Reply:
Do we need a separate test? This indirectly gets tested in other tests such as the test for insertion.

Reply:
In that case, it is fine ..